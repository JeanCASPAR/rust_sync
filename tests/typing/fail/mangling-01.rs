use rustre::sync;

sync! {
    node oui() = (a)
    where
        c : bool = false,
        a : int on c = 3 when c;

    node non() = ()
    where
        c : bool = true,
        a : int on c = oui(),
        l : int = merge c {
            true => a,
            false => 1 whennot c,
        };
}

fn main() {}
