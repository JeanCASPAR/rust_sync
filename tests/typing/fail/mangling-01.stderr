error: found type `int on c`, but a basic type was expected
 --> tests/typing/fail/mangling-01.rs:4:19
  |
4 |     node oui() = (a)
  |                   ^
